Remove unused argument to URLWithUserTypedString

LumioseSil opened the pull request:
#### 8e1bd72311a67d1d2e62dfb9c1e4b56798fb88d8
<pre>
Remove unused argument to URLWithUserTypedString
<a href="https://bugs.webkit.org/show_bug.cgi?id=252571">https://bugs.webkit.org/show_bug.cgi?id=252571</a>

URLWithUserTypedString is not used as a callback, and is a private function to WTF.

There is no benefit or need to have a second argument here if it is going to be ignored.

*Source\WTF\wtf\cocoa\NSURLExtras.h:
*Source\WTF\wtf\cocoa\NSURLExtras.mm:
*Tools\TestWebKitAPI\Tests\WTF\cocoa\URLExtras.mm:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8e1bd72311a67d1d2e62dfb9c1e4b56798fb88d8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/109047 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/18127 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/41868 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/580 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/118314 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/112932 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/19586 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/9429 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/101285 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/114806 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/14680 "Found 2 new test failures: fast/images/text-recognition/ios/show-data-detector-context-menu.html, system-preview/badge.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/97918 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/42818 "Found 4 new test failures: webgl/2.0.y/conformance/glsl/bugs/character-set.html, webgl/2.0.y/conformance/glsl/misc/fragcolor-fragdata-invariant.html, webgl/2.0.y/conformance/misc/invalid-passed-params.html, webgl/2.0.y/conformance/programs/program-test.html (failure)") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/96652 "3 api tests failed or timed out") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29568 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/84568 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/98082 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/10911 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/30915 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/98874 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/9003 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11651 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/7842 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/31020 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/17026 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50513 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/106543 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/13254 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/26406 "Passed tests") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

hortont424 reviewed (commented):
I have a strong hunch this was done in the interests of source compatibility with internal Apple code that uses WTF. May not be the case anymore, but this seems like a risky change unless coordinated with Apple folks.

LumioseSil committed:
Remove unused argument to URLWithUserTypedString
https://bugs.webkit.org/show_bug.cgi?id=252571

URLWithUserTypedString is not used as a callback, and is a private function to WTF.

There is no benefit or need to have a second argument here if it is going to be ignored.

*Source\WTF\wtf\cocoa\NSURLExtras.h:
*Source\WTF\wtf\cocoa\NSURLExtras.mm:
*Tools\TestWebKitAPI\Tests\WTF\cocoa\URLExtras.mm:
